lib/phoenix/socket/transport.ex

```diff
@@ -388,6 +388,57 @@ defmodule Phoenix.Socket.Transport do
     end
   end
 
+  @doc """
+  Extracts connection information from `conn` and returns a map.
+
+  Keys are retrieved from the optional transport option `:connect_info`.
+  This functionality is transport specific. Please refer to your transports's
+  documentation for more information.
+
+  The supported keys are:
+
+    * `:peer_data` - the result of `Plug.Conn.get_peer_data/1`.
+    * `:x_headers` - a list of all request headers that have an "x-" prefix.
+    * `:uri` - a `%URI{}` derived from the conn.
+
+  """
+  def connect_info(conn, opts) do
```

Comment:
Should the `connect_info` funcion receive only the `:connect_info` options instead of the whole `opts`? In other words, maybe `keys = Keyword.get(opts, :connect_info, opts)` should be a responsibility of the caller?

Reply:
Good catch. I believe this is also how other functions here behave.